torchaudio/_backend.py

```diff
@@ -1,43 +1,49 @@
-from typing import Any
+from typing import Any, Optional
 
-import platform
+from torchaudio._internal import module_utils as _mod_utils
+from . import _soundfile_backend, _sox_backend
 
+_BACKEND = None
+_BACKENDS = {}
 
-from . import _soundfile_backend, _sox_backend
+if _mod_utils.is_module_available('soundfile'):
+    _BACKENDS['soundfile'] = _soundfile_backend
+    _BACKEND = 'soundfile'
+if _mod_utils.is_module_available('torchaudio._torchaudio'):
+    _BACKENDS['sox'] = _sox_backend
+    _BACKEND = 'sox'
```

Comment:
nit: here, "sox" would become the default since it's the last one selected

Reply:
Yes, that's the intension but it was a bit unusual so I updated to use regular `if 'sox' else if 'soundfile'`, which is easier to read.